lib/engines/hcktest/hcktest.rb

```diff
@@ -225,22 +232,69 @@ def prepare_tests
         return
       end
 
-      @tests.list_tests(log: true)
+      @test_list = @tests.list_tests(log: true)
     end
 
     def auto_run
       ResourceScope.open do |scope|
-        run_and_configure_setup scope
-        prepare_tests
+        run_studio scope
+        sleep 5 until @studio.up?
 
-        @tests.run
+        run_tests
 
         pause_run if @project.options.test.manual
 
         @tests.create_project_package
       end
     end
 
+    def run_tests
+      run_normal_tests
+
+      group_tests.each do |group, tests|
+        next if tests.empty? || group == :normal
+
+        prepare_environment(group)
+        ResourceScope.open do |scope|
+          run_and_configure_clients scope
+
+          @tests.run(tests)
+          break if @project.run_terminated
+        end
+      end
+    end
```

Comment:
This is a bit strange division of concerns. There are two things in `run_tests`; running "normal" tests and running the other tests. However, while the first is abstracted away, the second isn't.

We can extract the second away by creating another method. Not being "normal" is "abnormal" but that sounds bad in English. Perhaps you may name it e.g., `run_tests_without_config` and `run_tests_with_config` to represent what it does more precisely; it is verbose but I think that's fine since it's only called from one place anyway.